File: IOS_swif_study/18NavigationBaritem_no_storyborard_obj_c/18NavigationBaritem_no_storyborard_obj_c/MainViewController.m
//
//  ViewController.m
//  18NavigationBaritem_no_storyborard_obj_c
//
//  Created by 성정현 on 2022/06/22.
//

#import "MainViewController.h"
#import "ProfileViewController.h"
#import "MessgeViewController.h"

@interface MainViewController ()

@end

@implementation MainViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    self.title = @"홈";
    self.view.backgroundColor = [UIColor systemRedColor];
    
    //  네비게이션 아이템 추가
    self.navigationItem.leftBarButtonItem =[[UIBarButtonItem alloc] initWithImage:[UIImage systemImageNamed:@"person.fill"] style:UIBarButtonItemStylePlain target:self action:@selector(goToProfileVC)];
    
    self.navigationItem.rightBarButtonItem =[[UIBarButtonItem alloc] initWithImage:[UIImage systemImageNamed:@"message.circle.fill"] style:UIBarButtonItemStylePlain target:self action:@selector(goToMessageVC)];
}

-(void)goToProfileVC{
    NSLog(@"MainViewController = goToProfileVC() called");
    ProfileViewController * profileVC = [[ProfileViewController alloc] init];
    [self.navigationController pushViewController:profileVC animated:YES];
    
}

-(void)goToMessageVC{
    NSLog(@"MainViewController = goToMessageVC() called");
    MessgeViewController * messageVC = [[MessgeViewController alloc] init];
    [self.navigationController pushViewController:messageVC animated:YES];
}

@end

File: IOS_swif_study/17_NavigationBaritem_noStory_swift_tutorial/17_NavigationBaritem_noStory_swift_tutorial/MainViewController.swift
//
//  ViewController.swift
//  17_NavigationBaritem_noStory_swift_tutorial
//
//  Created by 성정현 on 2022/06/05.
//

import UIKit

class MainViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        
        self.title = "홈"
        self.view.backgroundColor = .yellow
        
        //  네비게이션 아이템 추가
        navigationItem.leftBarButtonItem = UIBarButtonItem(image: UIImage(systemName: "person.fill"), style: .plain, target: self, action: #selector(goToProfileVC))
        navigationItem.rightBarButtonItem = UIBarButtonItem(image: UIImage(systemName: "message.fill"), style: .plain, target: self, action: #selector(goToMessageVC))
    }

    @objc fileprivate func goToProfileVC(){
        print("MainViewController - goTOProfileVC() 호출됨")
        let profileVC = ProfileViewController()
        self.navigationController?.pushViewController(profileVC, animated: true)
    }
    @objc fileprivate func goToMessageVC(){
        print("MainViewController - goToMessageVC() 호출됨")
        let messageVC = MessageViewController()
        self.navigationController?.pushViewController(messageVC, animated: true)
    }

}

